lib/models/project.js

```diff
@@ -416,9 +446,54 @@ class Project {
     @method discoverAddons
    */
   discoverAddons() {
-    let addonsList = this.addonDiscovery.discoverProjectAddons(this);
+    console.log(" ");
+    console.log(`discoverAddons for root ${this.root}, pkg ${this.pkg.name}.`);
+    let cache = this._packageInfoCache;
+
+    let projectPackageInfo = cache.getEntry(this.root);
+
+    let addonPackageList = [];
+
+    if (this.isEmberCLIAddon()) {
+        addonPackageList.push(projectPackageInfo);
+    }
+    
+    let cliInRepoAddons = (projectPackageInfo.cliEntry ? projectPackageInfo.cliEntry.inRepoAddons : null);
+    if (cliInRepoAddons) {
+      addonPackageList.push(...cliInRepoAddons);
+    }
+
+    if (projectPackageInfo.internalAddons) {
+      addonPackageList.push(...projectPackageInfo.internalAddons);
+    }
+
+    // The 'dependencyPackages' object contains ALL dependencies, not just those
+    // that are addons. We want only the addons
+    if (projectPackageInfo.dependencyPackages) {
+      let addonPkgs = Object.values(projectPackageInfo.dependencyPackages)
+          .filter(packageInfo => packageInfo.isAddon());
+      addonPackageList.push(...addonPkgs);
+    }
+
+    // The 'devDependencyPackages' object contains ALL devDependencies, not just those
+    // that are addons. We want only the addons.
+    if (projectPackageInfo.devDependencyPackages) {
+      let addonPkgs = Object.values(projectPackageInfo.devDependencyPackages)
+          .filter(packageInfo => packageInfo.isAddon());
+      addonPackageList.push(...addonPkgs);
+    }
+
+    if (projectPackageInfo.inRepoAddons) {
+      addonPackageList.push(...projectPackageInfo.inRepoAddons);
+    }
+
+    this.addonPackages = {};
 
-    this.addonPackages = this.addonDiscovery.addonPackages(addonsList);
+    addonPackageList.forEach(packageInfo => {
+      let relpath = path.relative(this.root, packageInfo.realPath);
+      console.log(`  ${packageInfo.name}: relpath: ${relpath}`);
```

Comment:
before merging, these logs will have to be removed, or switched to use heimdall-logger/debug